src/test/java/com/savvato/tribeapp/services/PhraseServiceImplTest.java

```diff
@@ -194,11 +193,41 @@ public void testApplyPhraseToUserWhenPhraseDoesNotExistInToBeReviewed() {
 
         Mockito.when(phraseRepository.findByAdverbIdAndVerbIdAndPrepositionIdAndNounId(any(Long.class), any(Long.class), any(Long.class), any(Long.class))).thenReturn(Optional.empty());
 
-        Mockito.when(toBeReviewedRepository.findByAdverbAndVerbAndNounAndPreposition(anyString(), anyString(), anyString(),anyString())).thenReturn(Optional.empty()).thenReturn(Optional.of(toBeReviewed));
+        Mockito.when(toBeReviewedRepository.findByAdverbAndVerbAndNounAndPreposition(anyString(), anyString(), anyString(),anyString())).thenReturn(Optional.of(toBeReviewed));
 
         boolean rtn = phraseService.applyPhraseToUser(user1.getId(), testWord, testWord, testWord, testWord);
 
         verify(reviewSubmittingUserRepository, times(1)).save(Mockito.any());
         assertFalse(rtn);
     }
+
+    @Test
+    public void testApplyPhraseToUserWhenAdverbIsBlank() {
+        User user1 = getUser1();
+        String testAdverb = "";
+        String testAdverbConverted = "nullvalue";
+        String testVerb = getTestVerb1().getWord();
+        String testPreposition = getTestPreposition1().getWord();
+        String testNoun = getTestNoun1().getWord();
+
+        ToBeReviewed tbrSaved = new ToBeReviewed();
+        tbrSaved.setId(1L);
+        tbrSaved.setHasBeenGroomed(false);
+        tbrSaved.setAdverb(testAdverbConverted);
+        tbrSaved.setVerb(testVerb);
+        tbrSaved.setPreposition(testPreposition);
+        tbrSaved.setNoun(testNoun);
+
+        // Should return false if the phrase has not been seen before
+        Mockito.when(toBeReviewedRepository.save(any())).thenReturn(tbrSaved);
+        assertFalse(phraseService.applyPhraseToUser(user1.getId(),testAdverb,testVerb,testPreposition, testNoun));
```

Comment:
Separate line 223 into 1) the call to `applyPhraseToUser` and 2) the assertion on its return value. The reason is that it's harder to see the 'when' (phraseService.applyPhraseToUser()) section of this test when it is wrapped in a 'then' function (assertFalse)

See https://martinfowler.com/bliki/GivenWhenThen.html

Reply:
separated